politeiawww/legacy/comments/process.go

```diff
@@ -314,26 +315,91 @@ func (c *Comments) processVotes(ctx context.Context, v v1.Votes) (*v1.VotesReply
 	cv := convertCommentVotes(votes)
 
 	// Populate comment votes with user data
-	uid, err := uuid.Parse(v.UserID)
+	err = c.commentVotesPopulateUserData(cv, v.UserID)
 	if err != nil {
 		return nil, err
 	}
-	u, err := c.userdb.UserGetById(uid)
-	if err != nil {
-		return nil, err
-	}
-	commentVotePopulateUserData(cv, *u)
-
-	// Sort comment votes by timestamp from newest to oldest.
-	sort.SliceStable(cv, func(i, j int) bool {
-		return cv[i].Timestamp > cv[j].Timestamp
-	})
 
 	return &v1.VotesReply{
 		Votes: cv,
 	}, nil
 }
 
+// usersBatchSize is the maximum number of users which can be fetched from
+// politeiawww and stored in memory while populating the comment votes structs
+// with the missing users data.
+var usersBatchSize = 10
+
+// commentVotePopulateUserData populates the comment votes with user data that
+// is not stored in politeiad. If all votes are associated with one user it
+// expects to get the user's ID as a parameter.
+func (c *Comments) commentVotesPopulateUserData(votes []v1.CommentVote, userID string) error {
+	// If given votes slice is emptry, nothing to do
+	if len(votes) == 0 {
+		return nil
+	}
+
+	// Collect the users public keys in a map to prevent duplicates and to
+	// retrieve the users in a batched db call.
+	var mPubKeys map[string]bool // map[pubkey]bool
+	if userID != "" {
+		// If user ID filter is applied, we have only one user
+		// to fetch.
+		mPubKeys = make(map[string]bool, 1)
+		mPubKeys[votes[0].PublicKey] = true
+	} else {
+		// If user ID filter is not applied, we need to collect all
+		// the user public keys from comment votes.
+		mPubKeys = make(map[string]bool, len(votes))
+		for _, vote := range votes {
+			if ok := mPubKeys[vote.UserID]; ok {
+				// If user uuid already known, skip
+				continue
+			}
+			mPubKeys[vote.PublicKey] = true
+		}
+	}
+
+	// Store public keys in a slice
+	pubKeys := make([]string, 0, len(mPubKeys))
+	for pubKey := range mPubKeys {
+		pubKeys = append(pubKeys, pubKey)
+	}
+
+	// Get users from db in batchs to avoid reading too many
+	// users into memory.
+	usernames := make(map[string]string, len(pubKeys))
+	for len(pubKeys) > 0 {
```

Comment:
Let's avoid using things like `math.Min` and floats unless there's a good reason to.  There's nothing inherently wrong with using them, there are just simplier ways to write this and the simplier solution is usually the better solution.

```
	var starIdx int
	for startIdx < len(pubkeys) {
		endIdx := startIdx + usersBatchSize
		if endIdx > len(pubkeys) {
			// We've reached the end of the slice
			endIdx = len(pubkeys)
		}

		// startIdx is included. endIdx is excluded.
		batch := pubkeys[startIdx:endIdx]

		// Do stuff

		// Update the start index
		startIdx = endIdx
	}
```

Reply:
like that, thanks